pkg/cmd/admin/validate/master.go

```diff
@@ -0,0 +1,172 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/util/fielderrors"
+
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	ValidateMasterConfigRecommendedName = "master-config"
+
+	validateMasterConfigLong = `
+Validate the configuration file for a master server.
+
+This command validates that a configuration file intended to be used for a master server is 
+created correctly. 
+`
+
+	valiateMasterConfigExample = ` // Validate master server configuration file
+  $ %s openshift.local.config/master/master-config.yaml`
+)
+
+type ValidateMasterConfigOptions struct {
+	// MasterConfigFile is the location of the config file to be validated
+	MasterConfigFile string
+
+	// Out is the writer to write output to
+	Out io.Writer
+}
+
+// NewCommandValidateMasterConfig provides a CLI handler for the `validate all-in-one` command
+func NewCommandValidateMasterConfig(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &ValidateMasterConfigOptions{
+		Out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SOURCE", name),
+		Short:   "Validate the configuration file for a master server",
+		Long:    validateMasterConfigLong,
+		Example: fmt.Sprintf(valiateMasterConfigExample, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(args, f); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			err, failure := options.Run()
+			if err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+			if failure {
+				options.Out.Write([]byte(fmt.Sprintf("FAILURE: Validation failed for file: %s\n", options.MasterConfigFile)))
+				os.Exit(1)
+			}
+
+			options.Out.Write([]byte(fmt.Sprintf("SUCCESS: Validation succeded for file: %s\n", options.MasterConfigFile)))
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *ValidateMasterConfigOptions) Complete(args []string, f *clientcmd.Factory) error {
+	switch len(args) {
+	case 0:
+		return errors.New("the path to the file being validated is required")
+	case 1:
+		o.MasterConfigFile = args[0]
+		return nil
+	default:
+		return errors.New("at most one file can be identified for validation")
```

Comment:
keep terminology consistent with the usage, like `only one source file is allowed`